### PR TITLE
Ci enhancement

### PR DESCRIPTION
**Pull request by TriplEight**
- show versions
- use `test.sh`
After https://github.com/paritytech/cargo-contract/pull/17 it will be possible to choose `cargo-contract` version.

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -10,13 +13,16 @@ if [ ! -f "$SUBSTRATE_PATH" ]; then
...
+
+echo "_____Purging dev chain_____"
 $SUBSTRATE_PATH purge-chain --dev -y
+echo "_____Spinning up the substrate node in background_____"
 $SUBSTRATE_PATH --dev &
 SUBSTRATE_PID=$!
 
-# Execute tests
+echo "_____Executing tests_____"
 yarn && yarn test
```
Comment (Stefie) [inline test.sh:27]: We could do 
`yarn upgrade --pattern @polkadot` 
to always test with the latest Polkadot JS API version.

Comment (TriplEight) [reply test.sh:27]: Done, thanks for the suggestion!

**Review by pepyakin — APPROVED** [review]

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -44,34 +45,28 @@ variables:
...
+    - rustup +nightly show
+    - cargo --version
+    - yarn --version
+    - npm --version
+    - wasm-prune -V
+    - sccache -s
     # `cargo install` returns an error if there is nothing to update, so have to supress it here temporarily
     - cargo contract -V
-    - cargo install --git https://github.com/paritytech/cargo-contract || echo $?
+    - cargo install --git https://github.com/paritytech/cargo-contract --branch master || echo $?
```
Comment (Robbepop) [inline .gitlab-ci.yml:57]: Can we instead use fixed releases?

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -9,15 +9,16 @@ stages:
...
 variables:
-  GIT_STRATEGY: fetch
-  GIT_SUBMODULE_STRATEGY: recursive
-  GIT_DEPTH: 3
-  CARGO_INCREMENTAL: 0
-  CARGO_HOME: "/ci-cache/${CI_PROJECT_NAME}/cargo/${CI_JOB_NAME}"
-  REGISTRY: registry.parity.io/parity/infrastructure/scripts
+  GIT_STRATEGY:                    fetch
+  GIT_SUBMODULE_STRATEGY:          recursive
+  GIT_DEPTH:                       3
```
Comment (Robbepop) [inline .gitlab-ci.yml]: Something magic about this git depth?

File: .gitlab-ci.yml (hunk 2 above)
Comment (TriplEight) [reply .gitlab-ci.yml:57]: of course, it's just not yet decided where to store them => where to take them from

File: .gitlab-ci.yml (hunk 3 above)
Comment (TriplEight) [reply .gitlab-ci.yml]: not at all, just faster to sync. Actually I will change it later.

Comment (Robbepop) [reply .gitlab-ci.yml]: but why not 1?

**Pull request merged**